experiments/sample_ssl4eo.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+""" Sample patch locations for downloading
+
+### run the script:
+## sample new locations with rtree overlap search
+python sample_ssl4eo.py \
+    --save-path ./data \
+    --size 1320 \
+    --num-cities 10000 \
+    --std 50 \
+    --indices-range 0 250000
+
+## resume from interruption
+python sample_ssl4eo.py \
+    --save-path ./data \
+    --size 1320 \
+    --num-cities 10000 \
+    --std 50 \
+    --resume ./data/sampled_locations.csv \
+    --indices-range 0 250000
+
+### Notes
+# The script will sample locations with rtree overlap search.
+# The script will save the sampled locations to a csv file.
+# By default, GaussianSampler is used to sample locations with a standard deviation
+ of 50 km from top 10000 populated cities.
+# Size (meter) is half the wanted patch size.
+
+"""
+
+import argparse
+import csv
+import os
+import time
+import warnings
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+from rtree import index
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+warnings.simplefilter("ignore", UserWarning)
+
+
+""" samplers to get locations of interest points"""
+
+
+def get_world_cities(download_root: str = "world_cities") -> pd.DataFrame:
+    url = "https://simplemaps.com/static/data/world-cities/basic/simplemaps_worldcities_basicv1.71.zip"  # noqa: E501
+    filename = "worldcities.csv"
+    if not os.path.exists(os.path.join(download_root, os.path.basename(url))):
+        download_and_extract_archive(url, download_root)
+    cities = pd.read_csv(os.path.join(download_root, filename))
+    cities.at[8436, "population"] = 50789  # fix one bug (Tecax) in the csv file
+    cities = cities[["city", "lat", "lng", "population"]]
```

Comment:
```suggestion
    cols = ["city", "lat", "lng", "population"]
    cities = pd.read_csv(os.path.join(download_root, filename), usecols=cols)
    cities.at[8436, "population"] = 50789  # fix one bug (Tecax) in the csv file
```
This will make reading faster